### ISSUE TITLE
Process exiting on errors in ExifTool

### ISSUE THREAD

**Issue by nknapp** [mceachen/exiftool-vendored.js#36]
**Describe the bug**

When I try to write tags to a non-existing file, the promise returned by `exiftool.write` 
does not get rejected. Instead, the whole process seems to be killed.

**To Reproduce**

The following code tries to write a tag to a file `abc.jpg` that we assume not to exist.

```js
const Exiftool = require("exiftool-vendored").ExifTool;
const exiftool = new Exiftool({});

(async () => {
  try {
    console.log("WRITING");
    await exiftool
      .write("abc.jpg", { AllDates: "2003-03-03-03-03-03" })
    console.log("DONE");
  } catch (err) {
    console.error("ERROR:", err);
  } finally {
    console.log("FINALLY");
  }
})();
```
**Expected behavior**

The promise returned by `exiftool.write` should be rejected, because the file `abc.jpg` does not exist.
The output should be 

```
WRITING
ERROR: <something about the missing file>
FINALLY
```

The actual output is: 

```
WRITING
```
which indicates that neither the catch-block, nor the line after the `.write()`, nor the finally block are executed.

**Environment (please complete the following information):**
 - Ubuntu 16.04.5 LTS
 - Node.JS version 8.12.0 and 10.11.0

**Comment by mceachen** [comment]
Thanks for reporting. 

I'm surprised this is an issue, as we already have a test for this exact case [here](https://github.com/mceachen/exiftool-vendored.js/blob/master/src/WriteTask.spec.ts#L122), unless I'm misunderstanding your problem.

**Comment by nknapp** [comment]
The tests all pass (142 total). After testing a little more, I found another hint.

When I add
```
setTimeout(() => console.log('TIMEOUT_CALLBACK'), 10000)
```
to the program, the error message is displayed. The problem is that my process exists too fast.
And the pending promise if the exiftool does not prevent the process from quitting.

**Comment by mceachen** [comment]
This seems to be an issue with `batch-cluster`: if I set the minor errors regex, things behave as expected:

```js
const exiftool = new ExifTool({ minorErrorsRegExp: /.*/ })
```
I'll look into more this tomorrow, but thanks again for reporting this.

**Comment by mceachen** [comment]
This will be addressed in version 7.0.0, which I hope to release tonight.
